Update/waku replace (#114)
status-im/specs: docs/stable/10-waku-mailserver.md

The envelope MUST be encrypted with a symmetric key agreed between the requester and the `Mailserver`.

### Receiving historic messages

Historic messages MUST be sent to a peer as a packet with a P2P Message code (`0x7f`) followed by an array of Waku envelopes.

In order to receive historic messages from a `Mailserver`, a node MUST trust the selected `Mailserver`, that is allow to receive packets with the P2P Message code. By default, such packets are discarded.

Received envelopes MUST be passed through the Waku envelope pipelines so that they are picked up by registered filters and passed to subscribers.

For a requester, to know that all messages have been sent by a `Mailserver`, it SHOULD handle P2P Request Complete code (`0x7d`). This code is followed by the following parameters:

```
[ RequestID, LastEnvelopeHash, Cursor ]
```

> `RequestID`: 32-byte wide array with a Keccak-256 hash of the envelope containing the original request  
> `LastEnvelopeHash`: 32-byte wide array with a Keccak-256 hash of the last sent envelope for the request  
> `Cursor`: an array of a cursor returned from the previous request (optional)

Reviewer: ```suggestion
 - `RequestID`: 32-byte wide array with a Keccak-256 hash of the envelope containing the original request  
 - `LastEnvelopeHash`: 32-byte wide array with a Keccak-256 hash of the last sent envelope for the request  
 - `Cursor`: an array of a cursor returned from the previous request (optional)
```

Author: Should the same change also be applied to `4-whisper-mailserver.md`?

Reviewer: probably, but not in this PR probably.

If `Cursor` is not empty, it means that not all messages were sent due to the set `Limit` in the request. One or more consecutive requests MAY be sent with `Cursor` field filled in order to receive the rest of the messages.

## Security considerations

### Confidentiality

All Waku envelopes are encrypted. A `Mailserver` node can not inspect their contents.

### Altruistic and centralized operator risk

> In order to be useful, a `Mailserver` SHOULD be online most of time. That means
> you either have to be a bit tech-savvy to run your own node, or rely on someone
> else to run it for you.

Reviewer: ```suggestion
In order to be useful, a `Mailserver` SHOULD be online most of time. That means
users either have to be a bit tech-savvy to run their own node, or rely on someone
else to run it for them.
```

Author: I thought we were addressing this type of thing in https://github.com/status-im/specs/issues/115
